File: message_queue_receiver.py
import sysv_ipc
key = sysv_ipc.ftok("msgqfile", 65)
queue = sysv_ipc.MessageQueue(key, sysv_ipc.IPC_CREAT | 0o666)
message, _ = queue.receive()
print(f"Message received: {message.decode()}")
queue.remove()

File: message_queue_receiver.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/ipc.h>
#include <sys/msg.h>
#include <sys/types.h>
struct msgbuf {
long mtype;
char mtext[100];
};
int main() {
key_t key;
int msgid;
key = ftok("msgqfile", 65);
msgid = msgget(key, 0666 | IPC_CREAT);
struct msgbuf message;
msgrcv(msgid, &message, sizeof(message), 1, 0);
printf("Message received: %s\n", message.mtext);
msgctl(msgid, IPC_RMID, NULL);
return 0;
}
